dotnet/SimpleSauce.Test/SauceOptionsTests.cs

```diff
@@ -132,5 +133,190 @@ public void AcceptsAllSauceLabsValues()
+            {
+                {"executable", "http://url.to/your/executable.exe"},
+                {"args", args},
+                {"background", false},
+                {"timeout", new TimeSpan(120)}
+            };
+
+            var tags = new List<string> { "foo", "bar" };
+
+            SauceOptions.AvoidProxy = true;
+            SauceOptions.BuildName = "Sample Build Name";
+            SauceOptions.CapturePerformance = true;
+            SauceOptions.ChromedriverVersion = "71";
+            SauceOptions.Timeout.CommandTimeout = 2;
+            SauceOptions.CustomData = customData;
+            SauceOptions.ExtendedDebugging = true;
+            SauceOptions.Timeout.IdleTimeout = 3;
+            SauceOptions.IeDriverVersion = "3.141.0";
+            SauceOptions.Timeout.MaxDuration = 300;
+            SauceOptions.TestName = "Sample Test Name";
+            SauceOptions.ParentTunnel = "Mommy";
+            SauceOptions.Prerun = prerun;
+            SauceOptions.Priority = 0;
+            SauceOptions.TestVisibility = TestVisibility.Team;
+            SauceOptions.RecordLogs = false;
+            SauceOptions.RecordScreenshots = false;
+            SauceOptions.RecordVideo = false;
+            SauceOptions.ScreenResolution = "10x10";
+            SauceOptions.SeleniumVersion = "3.141.59";
+            SauceOptions.Tags = tags;
+            SauceOptions.TimeZone = "San Francisco";
+            SauceOptions.TunnelIdentifier = "MyTunnel";
+            SauceOptions.VideoUploadOnPass = false;
+
+            var expectedCapabilities = new Dictionary<string, object>();
+
+            SauceOptions.ToDriverOptions().Should().BeEquivalentTo(expectedCapabilities);
+        }
+        [TestMethod]
+        public void ParsesCapabilitiesFromSeleniumValues()
+        {
+            var options = new ChromeOptions();
+            SauceOptions = new SauceOptions(options);
+
+            var expectedCapabilities = new Dictionary<string, object>();
+
+            SauceOptions.ToDriverOptions().Should().BeEquivalentTo(expectedCapabilities);
+        }
+
+        //TODO finish copying over this test from Java later
+        [TestMethod]
+        public void ParsesW3CAndSauceAndSeleniumValues()
+        {
+            var firefoxOptions = new FirefoxOptions();
+            firefoxOptions.AddArguments("--foo");
+            firefoxOptions.UnhandledPromptBehavior = UnhandledPromptBehavior.Dismiss;
+
+            SauceOptions = new SauceOptions(firefoxOptions)
+            {
+                PageLoadStrategy = PageLoadStrategy.Eager,
+                AcceptInsecureCerts = true,
+                AvoidProxy = true,
+                BuildName = "Sample Build Name",
+                CapturePerformance = true
+            };
+
+
+            var impl = new KeyValuePair<string, int>("implicit", 4);
+            var page = new KeyValuePair<string, int>("pageload", 44);
+            var script = new KeyValuePair<string, int>("script", 33);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'script'.](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4945851)